sourcecode/chesssimul/board.py

```diff
@@ -242,6 +245,48 @@ def move_bishop(self, move: Move) -> bool:
         else:
             return False
 
+    def move_rook(self, move: Move) -> bool:
+        iteration_lin: float = abs(move.end[0] - move.start[0])
+        iteration_col: float = abs(move.end[1] - move.start[1])
+
+        lin_sign: int = 1
+        col_sign: int = 1
+
+        # si le déplacement se fait bien en ligne droite
+        if move.end[0] != move.start[0] and move.end[1] == move.start[1] or move.end[0] == move.start[0] and move.end[1] != move.start[1]:
```

Comment:
I think you should add parenthesis around your and statements to make sure that they are executed before the or statement